resources/views/docs/1/digging-deeper/broadcasting.md

```diff
@@ -5,97 +5,82 @@ order: 100
-This allows you to listen for these events in real-time using Laravel Echo and react to these events on your application's
-front-end or via Laravel Livewire.
+## Custom events
 
-In order to make use of WebSockets in your NativePHP application, you can install the [Laravel WebSockets](https://beyondco.de/docs/laravel-websockets) package in your Laravel app.
-NativePHP automatically runs `php artisan websockets:serve` to start your WebSocket server once your application runs, allowing you to use Laravel Echo to connect to it and listen for events in real-time.
+You can also broadcast your own events. Simply implement the `ShouldBroadcastNow` contract in your event class and
+define the `broadcastOn` method, returning `nativephp` as one of the channels it broadcasts to:
 
-### Browser Events
+```php
+use Illuminate\Broadcasting\Channel;
+use Illuminate\Contracts\Broadcasting\ShouldBroadcastNow;
 
-In addition to connecting to a local WebSocket server using Laravel Echo, you can also listen to all internal PHP events in real-time using JavaScript by making use of Electron's [Inter-Process-Communication System](https://electronjs.org/docs/latest/api/ipc-renderer). 
-This is especially useful as it reduces the overhead and latency of dispatching and listening Laravel events via a WebSocket connection.
+class JobFinished implements ShouldBroadcastNow
+{
+    public function broadcastOn(): array
+    {
+        return [
+            new Channel('nativephp'),
+        ];
+    }
+}
+```
 
-### JavaScript
+This is particularly useful for scenarios where you want to offload an intensive task to a background queue and await
+its completion without constantly polling your application for its status.
 
-You can listen to these events in real-time by making use of Electron's `ipcRenderer` API.
-This API dispatches a `native-event` event, that contains the following payload:
+## Listening with JavaScript
 
-* `event` - A string containing the fully qualified class name of the NativePHP event class that got dispatched
-* `payload` - An object that contains the event payload
+You can listen to all native and custom events emitted by your application in real-time using JavaScript.
 
-```js
-import * as remote from '@electron/remote'
+NativePHP injecst a `window.Native` object into every window. The `on()` method allows you to register a callback as
+the second parameter that will run when the event specified in the first parameter is fired:
 
-ipcRenderer.on('native-event', (_, data) => {
-    const eventClassName = data.event;
-    const eventPayload = data.payload;
+```js
+Native.on("Native\\Laravel\\Events\\Windows\\WindowBlurred", (payload, event) => {
+    //
 });
 ```
 
-### Livewire
+Alternatively you can use the `Native.on` listener to immediately do something with the event payload. 
+
+``` js
+let podcastId = Native.on(
+    "App\\Events\\PodcastProcessed",
+    payload => payload.podcastId
+);
+```
```

Comment:
I'm not clear on this... the payload will only be available if the event fires.

I'm not sure we're returning anything from the `Native.on()` method right now... we return the result of the callback from the `ipcRenderer.on`, but we don't do anything with that

Am I missing something? 🤔

Reply:
No you are right. How embarrassing. 

I was burning the midnight oil yesterday. Was testing this in a console as a bare function (not inside that listener) got mixed up in my thinking. Should have re-checked before committing to the docs

Reply:
No problem at all! Don't be embarrassed. This is what review is for 👍🏼